vertical-pod-autoscaler/hack/vpa-process-yaml.sh

```diff
@@ -18,14 +18,27 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
-
 function print_help {
   echo "ERROR! Usage: vpa-process-yaml.sh <YAML files>+"
   echo "Script will output content of YAML files separated with YAML document"
   echo "separator and substituting REGISTRY and TAG for pod images"
 }
 
+function apply_feature_gate() {
+  filename=$1
+  if [ -n "${FEATURE_GATES}" ]; then
+    if [[ "${filename}" == *"admission-controller"* ]]; then
+      kubectl patch --type=json --local -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--feature-gates='"${FEATURE_GATES}"'"}]' -o yaml -f -
+    elif [[ "${filename}" == *"updater"* ]] || [[ "${filename}" == *"recommender"* ]]; then
+      kubectl patch --type=json --local -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args", "value": ["--feature-gates='"${FEATURE_GATES}"'"]}]' -o yaml -f -
+    else
+      cat # passthrough
```

Comment:
If this else statement matches, then the script hangs, waiting on cat to finish
```suggestion
```

Reply:
How are you calling the script? For me it works fine, but maybe I'm not covering some corner cases or something.

I've tried:
```bash
$ ./hack/vpa-process-yaml.sh ./deploy/vpa-v1-crd.yaml
$ FEATURE_GATES=InPlaceOrRecreate=true ./hack/vpa-process-yaml.sh ./deploy/vpa-v1-crd.yaml
$ FEATURE_GATES=InPlaceOrRecreate=true ./hack/vpa-process-yaml.sh ./deploy/admission-controller-deployment.yaml
$ FEATURE_GATES=InPlaceOrRecreate=true ./hack/vpa-process-yaml.sh ./deploy/updater-deployment.yaml
```
and I all get sensible outputs.

Reply:
I'm testing the unhappy paths. If one of these `else` statements were to match, the script will hang on `cat` running
I think they need to be removed, or exit with a warning